config/views/sitemaps.py

```diff
@@ -0,0 +1,124 @@
+from django.views.generic.base import TemplateView, TemplateResponseMixin
+from typing import Any
+from django.urls import reverse
+from ds_caselaw_utils import courts
+from caselawclient.Client import MarklogicResourceNotFoundError
+from caselawclient.client_helpers.search_helpers import (
+    search_judgments_and_parse_response,
+)
+from caselawclient.search_parameters import RESULTS_PER_PAGE, SearchParameters
+import datetime
+from typing import Union
+from judgments.utils import api_client
+
+
+class SitemapIndexView(TemplateView, TemplateResponseMixin):
+    content_type = "application/xml"
+    template_name = "sitemaps/index.xml"
+
+    def get_context_data(self, **kwargs) -> dict[str, Any]:
+        context = super().get_context_data(**kwargs)
+
+        # This is a list of the names of URLs for other sitemaps to appear in the index
+        map_url_names = [
+            "sitemap_static",
+            "sitemap_courts",
+        ]
+
+        # Build full URLs for all of the above, and put them in the context variable
+        context["maps"] = [self.request.build_absolute_uri(reverse(map)) for map in map_url_names]
+
+        # Dynamically append a sitemap for each court
+        for court in courts.get_listable_courts():
+            for year in range(court.start_year, court.end_year):
+                context["maps"].append(
+                    self.request.build_absolute_uri(
+                        reverse("sitemap_court", kwargs={"code": court.canonical_param, "year": year})
+                    )
+                )
+
+        return context
+
+
+class SitemapStaticView(TemplateView, TemplateResponseMixin):
+    content_type = "application/xml"
+    template_name = "sitemaps/sitemap.xml"
+
+    def get_context_data(self, **kwargs) -> dict[str, Any]:
+        context = super().get_context_data(**kwargs)
+
+        # This is a list of the names of URLs to reverse and include in the map
+        url_names = [
```

Comment:
Maybe not for now but it would be useful if we could somehow generate this from urls.py - I can see a case where we add a static page to the urls and forget to add it here.

Reply:
Agreed - static files are all over the place at the moment and it would be good to tidy them up.